lms/djangoapps/support/static/support/jsx/entitlements/components/Search/Search.jsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { InputText } from '@edx/paragon';
+
+
+class Search extends React.Component{
+  constructor(props) {
+    super(props)
+    this.state = { username: '' }
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    //since the below state update will cause react to re-render dom, the default refresh is unneeded
+    this.props.fetchEntitlements(this.state.username);
+  }
+
+  handleUserChange(event) {
```

Comment:
And, based on my reading of [the asInput Paragon component](https://github.com/edx/paragon/blob/3f31fa844fa28d936f6091400828f2a5bc2cf97c/src/asInput/index.jsx#L169), the callback isn't returning an event, but really the text value of the input, which is in this case, the username. So if you changed `event` to `username` you could do

```javascript
handleUserChange(username) {
  this.setState({ username });
}
```

Reply:
nice